assets/javascripts/lib/test-image-url.js.es6

```diff
@@ -1,20 +1,18 @@
-var testImageUrl = function(url) {
-  return new Promise(function (resolve, reject) {
-    let timeout = 5000;
-    let timer, img = new Image();
-    img.onerror = img.onabort = function () {
-      clearTimeout(timer);
-      resolve("error");
-    };
-    img.onload = function () {
-      clearTimeout(timer);
-      resolve("success");
-    };
-    timer = setTimeout(function () {
-      resolve("error");
-    }, timeout);
-    img.src = url;
-  });
-}
+var testImageUrl = function(url, callback) {
+  let timeout = 5000;
+  let timer, img = new Image();
+  img.onerror = img.onabort = function () {
+    clearTimeout(timer);
+    callback(false);
+  };
+  img.onload = function () {
+    clearTimeout(timer);
+    callback(true);
+  };
+  timer = setTimeout(function () {
+    callback(false);
+  }, timeout);
+  img.src = url;
```

Comment:
This is similar to the logic in the previous implementation, but I'm skeptical about it as I wonder if there's a danger of the img variable going out of scope and img.onload not being called? Despite this concern the function seems to work consistently in my testing